api/validator.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type address struct {
+	Host string
+	Port int
+}
+
+func (a address) String() string {
+	return fmt.Sprintf("%s:%d", a.Host, a.Port)
+}
+
+func (r Ingress) IsValid(cloudProvider string) error {
+	addrs := make(map[address]int)
+	nodePorts := make(map[int]int)
+	for ri, rule := range r.Spec.Rules {
+		if rule.HTTP != nil {
+			addr := address{Host: rule.Host}
+			if port, err := checkRequiredPort(rule.HTTP.Port); err != nil {
+				return fmt.Errorf("spec.rule[%d].http.port %s is invalid. Reason: %s", ri, rule.HTTP.Port, err)
+			} else {
+				addr.Port = port
+			}
+			if ei, found := addrs[addr]; found {
```

Comment:
i thik it is possible to reuse same host:port in two different rule in http mode.